internal/cluster/common.go

```diff
@@ -259,75 +259,45 @@ func MapHostsByStatus(c *common.Cluster) map[string][]*models.Host {
 	return mapHostsByStatus(c, "")
 }
 
-func UpdateMachineCidr(db *gorm.DB, cluster *common.Cluster, machineCidr string) error {
-	// In case of dual-stack clusters the autocalculation feature is not supported. That means
-	// as soon as we detect that current Machine Network configuration indicates we have such
-	// a cluster, the function stops its execution.
-	reqDualStack := network.CheckIfClusterIsDualStack(cluster)
-	if reqDualStack {
-		return nil
+func UpdateMachineCidr(db *gorm.DB, cluster *common.Cluster, machineCidr []string) error {
+	if len(machineCidr) > 2 {
+		return common.NewApiError(http.StatusInternalServerError,
+			errors.Errorf("for cluster %s received request to update %d machine networks", cluster.ID, len(machineCidr)))
```

Comment:
Users might have trouble understanding why this is a problem.
I'd change the error wording here to something like:
`Cannot update cluster %s machine networks. Supporting up to 2 networks (got %d)`

The `2` might be parameterized with a constant if you chose to, not a must.

Reply:
ok